### PR TITLE
ui: migrate cache:dir and check-ignore to ui.write

### PR DESCRIPTION
**Pull request by skshetry**
* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: tests/func/test_odb.py [hunk 1]
```diff
@@ -230,27 +229,3 @@ def test_shared_cache(tmp_dir, dvc, group):
     }
 
     assert expected == actual
-
-
-def test_cache_dir_local(tmp_dir, dvc, caplog):
```
Comment (skshetry) [inline tests/func/test_odb.py:235]: This test is not related to `odb` at all. It is related to the cache, but this got renamed, so this no longer belongs here.

**Review by pared — APPROVED** [review]

**Review by karajan1001 — APPROVED** [review]
LGTM
